_posts/news/2016-01-25-announcing-scalajs-0.6.6.md

```diff
@@ -0,0 +1,163 @@
+
+corresponds the JavaScript value
+
+    [45, "hello"]
+
+You should use `js.Tuple`s in facades to JavaScript libraries that take and return tuples as heterogeneous arrays.
+
+JavaScript tuples are convertible back and forth to Scala tuples, and you can use deconstruction to extract all values of a JavaScript at the same time.
+For example:
+
+    val scalaT: (Int, String) = t
+    val t2: js.Tuple2[Int, String] = scalaT
+    val js.Tuple2(a, b) = t2
+
+### Support for JUnit
+
+It is now possible to write JUnit tests for Scala.js!
+We have ported the entire JUnit API to Scala.js, so that it is now possible to write cross-compiling unit tests with JUnit.
+
+To enable JUnit on your Scala.js project, add the appropriate sbt AutoPlugin using
+
+    enablePlugins(ScalaJSJUnitPlugin)
+
+(like you would with the basic `ScalaJSPlugin`).
+For a `crossProject`, this must be enabled only on the JS variant.
+For the JVM variant, use the usual `libraryDependencies` setting:
+
+{% highlight scala %}
+lazy val myCrossProject = crossProject.
+  jsConfigure(_.enablePlugins(ScalaJSJUnitPlugin)).
+  jvmSettings(
+    libraryDependencies +=
+      "com.novocode" % "junit-interface" % "0.9" % "test"
+  )
+{% endhighlight %}
+
+### `js.ConstructorTag[C]`
+
+[`js.ConstructorTag[C]`]({{ site.production_url }}/api/scalajs-library/0.6.6/#scala.scalajs.js.ConstructorTag) is to [`js.constructorOf[C]`]({{ site.production_url }}/api/scalajs-library/0.6.6/#scala.scalajs.js.package@constructorOf[T<:scala.scalajs.js.Any]:scala.scalajs.js.Dynamic) as `ClassTag[C]` is to `classOf[C]`, i.e., you can use an `implicit` parameter of type `js.ConstructorTag[C]` to implicitly get a `js.constructorOf[C]`.
+For example:
+
+{% highlight scala %}
+def instantiate[C <: js.Any : js.ConstructorTag]: C =
+  js.Dynamic.newInstance(js.constructorTag[C].constructor)().asInstanceOf[C]
+  
+val newEmptyJSArray = instantiate[js.Array[Int]]
+{% endhighlight %}
+
+Implicit expansion will desugar the above code into:
+
+{% highlight scala %}
+def instantiate[C <: js.Any](implicit tag: js.ConstructorTag[C]): C =
+  js.Dynamic.newInstance(tag.constructor)().asInstanceOf[C]
+  
+val newEmptyJSArray = instantiate[js.Array[Int]](
+    new js.ConstructorTag[C](js.constructorOf[js.Array[Int]]))
+{% endhighlight %}
+
+although you cannot write the desugared version in user code because the constructor of `js.ConstructorTag` is private.
+
+This feature is particularly useful for Scala.js libraries wrapping JavaScript frameworks expecting to receive JavaScript constructors as parameters.
+
+### Query for Development versus Production mode
+
+A pair of methods allow you to query in your code whether the developer is doing a Development build or a Production build (for all practical purposes, aka `fastOptJS` and `fullOptJS`, resp.): `isDevelopmentMode` and `isProductionMode` in the object [`scala.scalajs.LinkingInfo`]({{ site.production_url }}/api/scalajs-library/0.6.6/#scala.scalajs.LinkingInfo$).
```

Comment:
This reads a bit hard. How about:

"The new methods `isDevelopmentMode` and `isProductionMode` allow you to query in your code whether you are running in a Development build or a Production build (for all ...). The methods are located in [`scala.scalajs.LinkingInfo`]({{ site.production_url }}/api/scalajs-library/0.6.6/#scala.scalajs.LinkingInfo$)."